src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,610 @@
+
+        $this->creatorFactory->addStmt(
+            new Node\Stmt\Return_(
+                new Node\Expr\Closure([
+                    'params' => [
+                        (new Param('configurator'))->setType('ContainerConfigurator')->getNode(),
+                    ],
+                    'stmts' => $closureStmt,
+                ])
+            )
+        );
+
+        $node = $this->creatorFactory->getNode();
+        $beforeClosure = !empty($this->useStatements) ? "\n" : null;
+        // add a blank line between the last use statement and the closure
+        array_splice($node->stmts,
+            -1,
+            0,
+            [new Node\Name($beforeClosure."/*\n * This file is the entry point to configure your own services.\n */")]
+        );
+
+        return (new Standard())->prettyPrintFile([$node])."\n";
+    }
+
+    /**
+     * @return Node[]
+     */
+    private function createClosureStmts(array $yamlData): array
+    {
+        foreach ($yamlData as $key => $values) {
+            if (null === $values) {
+                // if the key is written without values, we declare the variable ($parameters/$services).
+                $values = [];
+            }
+            switch ($key) {
+                case 'parameters':
+                    $this->addParametersNodes($values);
+                    break;
+                case 'imports':
+                    $this->addImportsNodes($values);
+                    break;
+                case 'services':
+                    $this->addServicesNodes($values);
+                    break;
+                default:
+                    throw new \LogicException(sprintf('The key %d is not available', $key));
+                    break;
+            }
+        }
+
+        sort($this->useStatements);
+        // Add at the end to sort them in alphabetical order.
+        foreach ($this->useStatements as $className) {
+            $this->creatorFactory->addStmt(
+                $this->factory->use($className)
+            );
+        }
+
+        // remove the last carriage return "\n" if exists.
+        $this->stmts[array_key_last($this->stmts)]->parts[0] = rtrim($this->stmts[array_key_last($this->stmts)]->parts[0], "\n");
+
+        return $this->stmts;
+    }
+
+    private function addParametersNodes(array $parameters)
+    {
+        $this->addLineStmt('// Put parameters here that don\'t need to change on each machine where the app is deployed.');
+        $this->addLineStmt('// https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration.');
+
+        $this->addLineStmt('$parameters = $configurator->parameters();', true);
+
+        foreach ($parameters as $parameterName => $value) {
+            $value = $this->toString($value, true);
+            $this->addLineStmt(sprintf(
+                '$parameters->set(%s, %s);',
+                $this->createStringArgument($parameterName),
+                $value
```

Comment:
I guess we could avoid the `$value = ` above and just put `$this->toString($value, true)` here

Reply:
Done.